numpy/ma/core.py

```diff
@@ -5057,7 +5057,7 @@ def mean(self, axis=None, dtype=None, out=None, keepdims=np._NoValue):
 
         if self._mask is nomask:
             result = super(MaskedArray, self).mean(axis=axis,
-                                                   dtype=dtype, **kwargs)
+                                                   dtype=dtype, **kwargs)[()]
         else:
```

Comment:
I assume the super mean returns a 0-d array scalar for relevant cases. I note the following odd behavior of ordinary arrays that is new to me.
```
>>> np.float64(1)
1.0
>>> np.float64(1)[()]
array(1.0)
>>> np.float64(1)[()][()]
1.0
```
That is, the `[()]` construct flips the result back and forth between arrays and scalars.